source/gba.cpp

```diff
@@ -14652,6 +14656,22 @@ int cheatsCheckKeys(u32 keys, u32 extended)
   return ticks;
 }
 
+
+
+void cheatListUpdate(int x) {
+	if(cheatsList[x].enabled)
+		sprintf(cheatsStringList[x], "%s (%s) Status: Enabled", cheatsList[x].desc, cheatsList[x].codestring);
+	else
+		sprintf(cheatsStringList[x], "%s (%s) Status: Disabled", cheatsList[x].desc, cheatsList[x].codestring);
+
+}
+
+void cheatListInit() {
+	for(int i = 0; i < 100; i++)
+		cheatsStringList[i] = (char*) malloc(100);
```

Comment:
please make this string buffer size a constant too

I can't find the place where you're freeing the memory, is it possible that you forgot to free the memory?

Reply:
Oh, that function was just supposed to be called once in the lifetime of the application, sorry for that. I'll move the call to cheatListInit away from the gba_init function.